evap/evaluation/tools.py

```diff
@@ -70,3 +73,100 @@ def set_content_disposition(self, filename):
+        "border_left_right":        xlwt.easyxf("borders: left medium, right medium"),
+        "border_top_bottom_right":  xlwt.easyxf("borders: top medium, bottom medium, right medium"),
+        "border_top":               xlwt.easyxf("borders: top medium"),
+    }
+
+    add_default_sheet = True
+    default_sheet_name = "EvaP Export"
+
+    def __init__(self):
+        self.workbook = xlwt.Workbook()
+        self.cur_row = 0
+        self.cur_col = 0
+        if self.add_default_sheet:
+            self.cur_sheet = self.workbook.add_sheet(self.default_sheet_name)
+        else:
+            self.cur_sheet = None
+        self.styles = self.init_styles()
+
+    def write(self, label="", style_name="default", rows=1, cols=1):
+        """Write whatever you want. Consider the other write_* methods."""
+        style = self.styles[style_name]
+        if rows > 1 or cols > 1:
+            self.cur_sheet.write_merge(
+                self.cur_row, self.cur_row + rows - 1, self.cur_col + cols - 1, style)
+            self.cur_col += cols - 1
+        else:
+            self.cur_sheet.write(self.cur_row, self.cur_col, label, style)
+
+    def write_cell(self, label="", style_name="default"):
+        """Write a single cell and move to the next column."""
+        self.write(label, style_name)
+        self.cur_col += 1
+
+    def next_row(self):
+        self.cur_col = 0
+        self.cur_row += 1
+
+    def write_cell_on_next_row(self, label="", style_name="default"):
+        """Write the cell at the beginning of the next row."""
+        self.next_row()
+        self.write_cell(label, style_name)
+
+    def write_row(self, *vals, style="default"):
+        """
+        Write a cell for every value in the *args and go to the next row.
+        Styling can be chosen
+          - once for all cells by providing a str
+          - separately for the first n cells by providing a list of length n
+            (the other cells will have the style "default")
+          - not at all and defaulted.
+        """
+        if isinstance(style, str):
+            styles = itertools.repeat(style)
+        else:
+            styles = itertools.chain(style, itertools.repeat("default"))
+
+        for val, cell_style in zip(vals, styles):
+            self.write_cell(val, cell_style)
+        self.next_row()
+
+    def write_empty_row_with_styles(self, styles):
+        for style in styles:
+            self.write_cell(None, style)
+        self.next_row()
+
+    def write_row_with_style_mapping(self, row_values, map_style):
+        for val in row_values:
+            self.write_cell(val, style_name=map_style(val))
+        self.next_row()
+
+    @classmethod
+    def init_styles(cls):
+        return cls.default_styles.copy()
+
+    def create_color(self, color_name, palette_index, color):
+        xlwt.add_palette_colour(color_name, palette_index)
+        self.workbook.set_colour_RGB(palette_index, *color)
```

Comment:
since this relies on the CUSTOM_COLOR_START stuff to work, i would either move this method to the one exporter that needs it, or move the palette_index logic into this class here (keeping a counter of the already taken palette indices, starting with CUSTOM_COLOR_START )